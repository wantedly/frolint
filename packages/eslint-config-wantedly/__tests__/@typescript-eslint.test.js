const path = require("path");
const CLIEngine = require("eslint").CLIEngine;

const ESLINT_CONFIG_FILE = path.resolve(
  __dirname,
  "..",
  "@typescript-eslint.js"
);
const engine = new CLIEngine({
  configFile: ESLINT_CONFIG_FILE,
  useEslintrc: false,
});

describe("eslint-config-wantedly/@typescript-eslint", () => {
  test("should match snapshot", () => {
    expect(engine.config.specificConfig).toMatchSnapshot();
  });
});
