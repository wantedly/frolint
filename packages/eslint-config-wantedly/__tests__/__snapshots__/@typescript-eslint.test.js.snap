// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-wantedly/@typescript-eslint should match snapshot 1`] = `
Object {
  "baseDirectory": "/Users/yamadayuki/dev/src/github.com/yamadayuki/eslint-config-wantedly",
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "prettier/@typescript-eslint",
  ],
  "filePath": "/Users/yamadayuki/dev/src/github.com/yamadayuki/eslint-config-wantedly/@typescript-eslint.js",
  "globals": Object {},
  "parser": "/Users/yamadayuki/dev/src/github.com/yamadayuki/eslint-config-wantedly/node_modules/@typescript-eslint/parser/dist/parser.js",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "react",
    "@typescript-eslint",
    "prettier",
  ],
  "rules": Object {
    "@typescript-eslint/adjacent-overload-signatures": "error",
    "@typescript-eslint/array-type": "error",
    "@typescript-eslint/ban-types": "error",
    "@typescript-eslint/camelcase": "error",
    "@typescript-eslint/class-name-casing": "error",
    "@typescript-eslint/explicit-function-return-type": Array [
      "off",
    ],
    "@typescript-eslint/explicit-member-accessibility": "error",
    "@typescript-eslint/indent": "off",
    "@typescript-eslint/interface-name-prefix": "error",
    "@typescript-eslint/member-delimiter-style": "off",
    "@typescript-eslint/no-angle-bracket-type-assertion": "error",
    "@typescript-eslint/no-array-constructor": "error",
    "@typescript-eslint/no-empty-interface": "error",
    "@typescript-eslint/no-explicit-any": Array [
      "off",
    ],
    "@typescript-eslint/no-inferrable-types": "error",
    "@typescript-eslint/no-misused-new": "error",
    "@typescript-eslint/no-namespace": "error",
    "@typescript-eslint/no-non-null-assertion": "error",
    "@typescript-eslint/no-object-literal-type-assertion": "error",
    "@typescript-eslint/no-parameter-properties": "error",
    "@typescript-eslint/no-triple-slash-reference": "error",
    "@typescript-eslint/no-unused-vars": Array [
      "error",
      Object {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_",
      },
    ],
    "@typescript-eslint/no-use-before-define": "error",
    "@typescript-eslint/no-var-requires": "error",
    "@typescript-eslint/prefer-interface": "error",
    "@typescript-eslint/prefer-namespace-keyword": "error",
    "@typescript-eslint/type-annotation-spacing": "off",
    "accessor-pairs": "off",
    "array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "array-callback-return": "off",
    "array-element-newline": "off",
    "arrow-body-style": 0,
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "block-scoped-var": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "callback-return": "off",
    "camelcase": "off",
    "capitalized-comments": "off",
    "class-methods-use-this": "off",
    "comma-dangle": "off",
    "comma-spacing": "off",
    "comma-style": "off",
    "complexity": "off",
    "computed-property-spacing": "off",
    "consistent-return": "off",
    "consistent-this": "off",
    "constructor-super": "error",
    "curly": 0,
    "default-case": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "eol-last": "off",
    "eqeqeq": "off",
    "for-direction": "error",
    "func-call-spacing": "off",
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": "off",
    "function-paren-newline": "off",
    "generator-star": "off",
    "generator-star-spacing": "off",
    "getter-return": "error",
    "global-require": "off",
    "guard-for-in": "off",
    "handle-callback-err": "off",
    "id-blacklist": "off",
    "id-length": "off",
    "id-match": "off",
    "implicit-arrow-linebreak": "off",
    "indent": "off",
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": "off",
    "line-comment-position": "off",
    "linebreak-style": "off",
    "lines-around-comment": 0,
    "lines-around-directive": "off",
    "lines-between-class-members": "off",
    "max-classes-per-file": "off",
    "max-depth": "off",
    "max-len": 0,
    "max-lines": "off",
    "max-lines-per-function": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "off",
    "multiline-comment-style": "off",
    "multiline-ternary": "off",
    "new-cap": "off",
    "new-parens": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": "off",
    "no-array-constructor": "off",
    "no-arrow-condition": "off",
    "no-async-promise-executor": "off",
    "no-await-in-loop": "off",
    "no-bitwise": "off",
    "no-buffer-constructor": "off",
    "no-caller": "off",
    "no-case-declarations": "error",
    "no-catch-shadow": "off",
    "no-class-assign": "error",
    "no-comma-dangle": "off",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": 0,
    "no-console": "warn",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "warn",
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-eq-null": "off",
    "no-eval": "off",
    "no-ex-assign": "error",
    "no-extend-native": "off",
    "no-extra-bind": "off",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "off",
    "no-extra-parens": "off",
    "no-extra-semi": "off",
    "no-fallthrough": "error",
    "no-floating-decimal": "off",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implicit-coercion": "off",
    "no-implicit-globals": "off",
    "no-implied-eval": "off",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": "off",
    "no-magic-numbers": "off",
    "no-misleading-character-class": "off",
    "no-mixed-operators": 0,
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-assign": "off",
    "no-multi-spaces": "off",
    "no-multi-str": "off",
    "no-multiple-empty-lines": "off",
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-negated-in-lhs": "off",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-object": "off",
    "no-new-require": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": "off",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "off",
    "no-prototype-builtins": "off",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-reserved-keys": "off",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-properties": "off",
    "no-restricted-syntax": "off",
    "no-return-assign": "off",
    "no-return-await": "off",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "off",
    "no-sequences": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": "off",
    "no-space-before-semi": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-tabs": 0,
    "no-template-curly-in-string": "off",
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-throw-literal": "off",
    "no-trailing-spaces": "off",
    "no-undef": "error",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unexpected-multiline": 0,
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-vars": Array [
      "error",
      Object {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_",
      },
    ],
    "no-use-before-define": "off",
    "no-useless-call": "off",
    "no-useless-catch": "off",
    "no-useless-computed-key": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "error",
    "no-useless-rename": "off",
    "no-useless-return": "off",
    "no-var": "off",
    "no-void": "off",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "off",
    "no-with": "off",
    "no-wrap-func": "off",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": "off",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "off",
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "padding-line-between-statements": "off",
    "prefer-arrow-callback": 0,
    "prefer-const": "off",
    "prefer-destructuring": "off",
    "prefer-numeric-literals": "off",
    "prefer-object-spread": "off",
    "prefer-promise-reject-errors": "off",
    "prefer-reflect": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "prettier/prettier": "error",
    "quote-props": "off",
    "quotes": 0,
    "radix": "off",
    "react/display-name": 2,
    "react/jsx-key": 2,
    "react/jsx-no-comment-textnodes": 2,
    "react/jsx-no-duplicate-props": 2,
    "react/jsx-no-target-blank": 2,
    "react/jsx-no-undef": 2,
    "react/jsx-uses-react": Array [
      "error",
    ],
    "react/jsx-uses-vars": "error",
    "react/no-children-prop": 2,
    "react/no-danger-with-children": 2,
    "react/no-deprecated": 2,
    "react/no-direct-mutation-state": 2,
    "react/no-find-dom-node": 2,
    "react/no-is-mounted": 2,
    "react/no-render-return-value": 2,
    "react/no-string-refs": 2,
    "react/no-unescaped-entities": 2,
    "react/no-unknown-property": 2,
    "react/no-unsafe": 0,
    "react/prop-types": 2,
    "react/react-in-jsx-scope": 2,
    "react/require-render-return": 2,
    "require-atomic-updates": "off",
    "require-await": "off",
    "require-jsdoc": "off",
    "require-unicode-regexp": "off",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "semi": "off",
    "semi-spacing": "off",
    "semi-style": "off",
    "sort-imports": "off",
    "sort-keys": "off",
    "sort-vars": "off",
    "space-after-function-name": "off",
    "space-after-keywords": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-before-function-parentheses": "off",
    "space-before-keywords": "off",
    "space-in-brackets": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-return-throw-case": "off",
    "space-unary-ops": "off",
    "space-unary-word-ops": "off",
    "spaced-comment": "off",
    "strict": "off",
    "switch-colon-spacing": "off",
    "symbol-description": "off",
    "template-curly-spacing": "off",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-jsdoc": "off",
    "valid-typeof": "error",
    "vars-on-top": "off",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "yoda": "off",
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
